src/sysext/sysext.c

```diff
@@ -273,6 +273,106 @@ static int need_reload(
         return false;
 }
 
+static int move_submounts(const char *src, const char *dst, const char *stack) {
+        SubMount *submounts = NULL;
+        size_t n_submounts = 0;
+        int r = 0;
+
+        assert(src);
+        assert(dst);
+        assert(stack);
+
+        CLEANUP_ARRAY(submounts, n_submounts, sub_mount_array_free);
+
+        r = get_sub_mounts(src, &submounts, &n_submounts);
```

Comment:
i would feel better if we'd first create a local copy of the /etc/ tree, then turn off propagation from our side (i.e. MS_SLAVE), before we peal anything off the tree.

Reply:
hmm, to be fully correct this should probably a loop, so that we can deal with people adding more mounts while we unpeal the tree

Reply:
A loop seems like it would be rather complex since we'd also need to handle unmounts then, not just more mounts being added.

Reply:
@poettering How would `MS_SLAVE` help here? We actually want to unmount in the host mount namespace.

Reply:
so i think in the long run we should not unmount in this loop really, but instead leave everything in place until we are fully done, and then insert the full new mount tree with the beaneath mount logic, then reveal it by MS_MOVEing the old tree away, and then unmounting the old tree recursively.